src/cca/ui_components/server_card.ts

```diff
@@ -0,0 +1,113 @@
+import { Server } from '../model/server';
+
+enum State {
+  DISCONNECTED,
+  CONNECTING,
+  CONNECTED,
+  DISCONNECTING
+}
+
+// An entry in a ServerList.
+export class ServerCard {
+  private state = State.DISCONNECTED;
+
+  private button: HTMLButtonElement;
+  private status: HTMLElement;
+
+  constructor(private root: HTMLElement, private server: Server) {
+    root.classList.add('server-card');
+    root.innerHTML = `
+      <div class='toolbar'>
+        <div class='paper-font-title'>${server.getIpAddress()}</div>
```

Comment:
You can use:
 
    root.querySelector('.paper-font-title').textContent = server.getIpAddress();

This way there's no risk for script injection.